### PR TITLE
[FEAT] Eventsim to support publishing data HPCM kafka

### PR DESCRIPTION
**Pull request by sindhu-106**
Signed-off-by: sindhuja <sindhuja.gajula@intel.com>

### CODE REVIEW

File: configurations/eventsim/EventSim.json [hunk 1]
```diff
@@ -18,8 +15,9 @@
     "time-delay-mus": 1,
     "timezone": "GMT"
   },
-  "network-config" : {
-      "network" : "sse",
+  "server-network-config" : {
+      "server-network" : "sse",
+      "publisher-network": "sse",
```
Comment (paul-amonson) [inline configurations/eventsim/EventSim.json:20]: If using Kafka why is this here?

File: eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java [hunk 2]
```diff
@@ -245,7 +245,7 @@ private void publishEchoEvent ( PropertyMap event, String streamID ) throws Simu
     }
 
     private PropertyDocument generateEvents(Map<String, String> parameters, String type) throws SimulatorException, PropertyNotExpectedType, IOException, ConfigIOParseException, ConversionException {
-        DataValidation.validateData(parameters, MISSING_KEY);
+        DataValidation.isDataNullOrEmpty(parameters, MISSING_KEY);
```
Comment (paul-amonson) [inline eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java:248]: Much better name!  :)

File: eventsim/src/main/java/com/intel/dai/eventsim/NetworkConnectionObject.java [hunk 3]
```diff
@@ -4,8 +4,12 @@
 import com.intel.networking.restclient.RESTClientException;
 import com.intel.networking.restserver.RESTServerException;
 import com.intel.networking.restserver.RESTServerHandler;
+import com.intel.networking.source.NetworkDataSource;
+import com.intel.networking.source.NetworkDataSourceEx;
+import com.intel.networking.source.NetworkDataSourceFactory;
 import com.intel.properties.PropertyDocument;
 import com.intel.properties.PropertyMap;
+import org.apache.commons.collections4.map.HashedMap;
```
Comment (paul-amonson) [inline eventsim/src/main/java/com/intel/dai/eventsim/NetworkConnectionObject.java:12]: Was this intentional or did you actually want to use the built in HashMap?

File: eventsim/src/main/java/com/intel/dai/eventsim/NetworkConnectionObject.java [hunk 4]
```diff
@@ -17,24 +21,56 @@
...
-            return callBack;
+        if (publisher.equals(NETWORK_TYPES.callback)) {
+            publisher_ = callBack;
+            return;
         }
 
-        if (NETWORK_TYPES.valueOf(network.toUpperCase()) == NETWORK_TYPES.RABBITMQ) {
-            return null;
-            //TO-DO Implementation of rabbitmq
+        Map<String, String> convertedConfig = new HashedMap<>();
```
Comment (paul-amonson) [inline eventsim/src/main/java/com/intel/dai/eventsim/NetworkConnectionObject.java:55]: Same comment: Apache's HashedMap or built in HashMap?

File: eventsim/src/main/java/com/intel/dai/eventsim/NetworkConnectionObject.java [hunk 5]
```diff
@@ -179,13 +226,39 @@ public PropertyDocument getAllSubscriptions() {
...
+    }
+
     enum NETWORK_TYPES {
-        SSE,
-        RABBITMQ,
-        CALLBACK,
-        OTHER
+        sse,
+        rabbitmq,
+        callback,
```
Comment (paul-amonson) [inline eventsim/src/main/java/com/intel/dai/eventsim/NetworkConnectionObject.java:252]: As a general code comment, you may wish to remove all references to the HTTP callback method as I have already removed the other code outside EventSim. :) It should be removed here but not necessarily in this review.

File: eventsim/src/main/java/com/intel/dai/eventsim/NetworkConnectionObject.java (hunk 3 above)
Comment (sindhu-106) [reply eventsim/src/main/java/com/intel/dai/eventsim/NetworkConnectionObject.java:12]: Good catch Paul, Its IDE import error. I want to use HashMap only :)

File: configurations/eventsim/EventSim.json (hunk 1 above)
Comment (sindhu-106) [reply configurations/eventsim/EventSim.json:20]: Our nightly tests machine do not support kafka. So temporarily we should use sse to support current platform.

File: eventsim/src/main/java/com/intel/dai/eventsim/NetworkConnectionObject.java (hunk 5 above)
Comment (sindhu-106) [reply eventsim/src/main/java/com/intel/dai/eventsim/NetworkConnectionObject.java:252]: sure Paul we will do it in a separate review

**Review by paul-amonson — APPROVED** [review]

**Review by nsai1 — APPROVED** [review]
